src/util/bindProps.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { forwardRef } from 'react';
+import type {
+  ComponentType,
+  ComponentPropsWithoutRef,
+  ElementRef,
+  ElementType,
+  ForwardedRef,
+} from 'react';
+import type { OmitByType } from './typeHelpers';
+
+/**
+ * Bind props to a component.
+ *
+ * const PasswordInput = bindProps(Input, { type: 'password' });
+ * <PasswordInput ...any Input props... />
+ */
+export function bindProps<C extends ElementType, P extends Partial<ComponentPropsWithoutRef<C>>>(
+  Component: C,
+  boundProps: P
+) {
+  type OProps = ComponentPropsWithoutRef<C>;
+  type ORefInstance = ElementRef<C>;
+  type BProps = OmitByType<P, undefined>;
+  const Comp = Component as ComponentType<OProps>;
+
+  return forwardRef(
+    (props: Omit<OProps, keyof BProps> & Partial<BProps>, ref: ForwardedRef<ORefInstance>) => {
+      const allProps = { ...boundProps, ...props } as OProps;
+      return <Comp {...allProps} ref={ref} />;
```

Comment:
Most component in reactstrap don't actually have a ref prop. It usually uses `innerRef`

Reply:
`ref` will be undefined if not used so it shouldn't hurt. `innerRef` will be passed in through normal props.